lib/deimos/active_record_consume/batch_record.rb

```diff
@@ -26,7 +26,7 @@ def initialize(klass:, attributes:, bulk_import_column: nil)
         @klass = klass
         if bulk_import_column
           self.bulk_import_column = bulk_import_column
-          self.bulk_import_id = SecureRandom.uuid
+          self.bulk_import_id = ULID.generate
```

Comment:
https://github.com/flipp-oss/deimos/blob/master/lib/deimos/active_record_consume/batch_record_list.rb#L47 

This section uses `select * from table where bulk_import_id in (....)` . This is super slow for indexed GUID columns when table size is >300k rows. ULIDs bring in huge benefit in this context.

Do I consider making ULID part of deimos dependency OR should I define a method for calling projects to override?

Reply:
I'd rather not add a dependency. Providing a method to generate the ID is better IMO.